Creates Julia module for the project

exoticDFT opened the pull request:
- Creates a Julia module for the project to call externally in other Julia code.
  - Removes runtime code from the converter file.
  - Adds Project.toml file for creating the module.
  - Adds "main" project Julia file for the module.

PeggyWang0 committed:
dev-ing

PeggyWang0 committed:
new tests

PeggyWang0 committed:
update julia notebook

PeggyWang0 committed:
working 2 lane rd

PeggyWang0 committed:
Julia File for converter

exoticDFT committed:
Updates project to start packing for external use

- Adds Project.toml to load package outside of local development.
- Adds main Julia file for exporting functions for external use.
- Comments out test code in converter.jl so the module can load only the functions when imported.

PeggyWang0 committed:
Fixed segid bug

PeggyWang0 committed:
Segid small changes

exoticDFT committed:
Merge remote-tracking branch 'origin/converterDev' into packaging

- Pulls in bug-fix for roadway converter.

PeggyWang0 committed:
Fixed segid bug again

exoticDFT committed:
Updates xodr to roadway converter module

- Updates the createCurve function to handle roads with arbitrary line geometry.
  - Updates function to take in road geometry and lane data structures.
  - Allows any orientation of road segment to be generated.
  - Creates location of curve points that match the center of the lanes in OpenDrive.
- Updates the OpenDriveToRoadwaysConverter function to handle new parameters for createCurve function.

exoticDFT committed:
Merge branch 'converterDev' into packaging

exoticDFT committed:
Removes runtime code from converter module

- Removes the commented code from the converter module as this is should be moved to a text or other runtime file.

exoticDFT commented:
Created the PR, but should wait to review and merge until after we finalize PR #4

PeggyWang0 commented on src/OpenDrive2Roadway.jl:
```diff
@@ -0,0 +1,9 @@
+module OpenDrive2Roadway
```
I think we should call it something like OpenDriveToRoadway rather than with the numerical 2 (for readability, numerals are usually for versions)

exoticDFT replied:
Sounds good, I'll update this.

PeggyWang0 commented on src/converter.jl:
```diff
@@ -0,0 +1,66 @@
+function createCurve(segid, laneindex, road_geom, lane)
+    tag = LaneTag(segid, laneindex)
+    curve = nothing
+        if road_geom.type_name == "line"
```
Should this be indented? It's indented right now but this "if" statement should be indented back one no?

exoticDFT replied:
Not sure how this really happened. Definitely no need for the extra indent :-p

PeggyWang0 reviewed (changes requested):
See comments

PeggyWang0 commented on main.py:
```diff
@@ -9,9 +9,10 @@
-        onlyfiles.remove(".DS_Store")
+    # onlyfiles = [f for f in listdir(args.folder) if isfile(join(args.folder, f))]
+    # if ".DS_Store" in onlyfiles:
+    #     onlyfiles.remove(".DS_Store")
     odp.parse_file(args.filename)
-    for f in onlyfiles:
-        odp.parse_file(args.folder+f)
+    print(odp.data.roads[1].lanes.laneSection[0])
+    # for f in onlyfiles:
+    #     odp.parse_file(args.folder+f)
```
Uncomment these lines

exoticDFT replied:
Is this just a Mac specific thing? Does it affect the actual running of the code at all? I don't think there is any harm in keeping it, but I think I commented this out because I thought it didn't really influence the functionality. Although I'm not sure why I didn't just delete it.

exoticDFT committed:
Fixes some comments for PR #5

- Updates the formatting within a function in converter Julia file.

exoticDFT committed:
Renames the Julia module to OpenDriveToRoadway

- Updates the name of the Julia module to OpenDriveToRoadway in the Project.toml.
- Renames the "main" Julia file to match the module name change.

exoticDFT committed:
Reinstates some Mac specific code

- Uncomments code that was commented for non-Mac systems.

PeggyWang0 reviewed (approved):
Amazing :) 👍
